### PR TITLE
feat: add process metrics

### PR DESCRIPTION
**Pull request by gshilei**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):
-  Y 
feat: [Kusion module should have better panic and error handling and support trace and metrics](https://github.com/KusionStack/kusion/issues/1305)

#### 2. What is the scope of this PR (e.g. component or file name):
kusion-module-framework

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):
Add process metrics package

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):
- N

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:
- Unit test

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/metrics/process.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+		}
+	}
+	customRegistry := prometheus.NewRegistry()
+	customRegistry.MustRegister(ProcessCPUPercent)
+	customRegistry.MustRegister(ProcessMemoryPercent)
+	customRegistry.MustRegister(ProcessMemoryRSS)
+	customRegistry.MustRegister(ProcessRequestDuration)
+	customRegistry.MustRegister(ProcessPerRequestDuration)
+
+	var errCH chan error
```
Comment (liu-hm19) [inline pkg/metrics/process.go]: Here seems to be a typo? According to the `camelCase` naming convention, it should be `errCh` here? 😂

File: pkg/metrics/process.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to get process[%v] parent process pid: %v", pid, err)
+	}
+	cpuPercent, err := p.CPUPercent()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get process[%v] cpu percent: %v", pid, err)
+	}
+	memoryPercent, err := p.MemoryPercent()
+	if err != nil {
+		fmt.Errorf("failed to get process[%v] memory percent: %v", pid, err)
```
Comment (liu-hm19) [inline pkg/metrics/process.go]: The result of `fmt.Errorf` seems not used, should it be `return nil, fmt.Errorf()`?

File: pkg/metrics/process.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+	}
+	memInfo, err := p.MemoryInfo()
+	if err != nil {
+		fmt.Errorf("failed to get process[%v] memory info: %v", pid, err)
```
Comment (liu-hm19) [inline pkg/metrics/process.go]: Ditto.

File: pkg/metrics/process.go (hunk 1 above)
Comment (gshilei) [reply pkg/metrics/process.go]: fixed

File: pkg/metrics/process.go (hunk 2 above)
Comment (gshilei) [reply pkg/metrics/process.go]: fixed

File: pkg/metrics/process.go (hunk 3 above)
Comment (gshilei) [reply pkg/metrics/process.go]: fixed

**Review by liu-hm19 — APPROVED** [review]
lgtm

**Pull request merged**